src/Entity.php

```diff
@@ -41,19 +47,23 @@ public function __construct($entity, $id = null)
      */
     public function &__get($name)
     {
-        $names = array(
-            'entity' => 'entity',
-            'id'     => 'id',
-        );
-        $name = strtolower($name);
-        if (!in_array($name, array_keys($names))) {
-            throw new Exception\InvalidArgumentException(sprintf(
-                'Invalid property name "%s"',
-                $name
-            ));
-        }
-        $prop = $names[$name];
-        return $this->{$prop};
+        throw new \Exception('Direct query of values is deprecated.  Use getters.');
```

Comment:
We can make this BC by doing the following:

- Make `getEntity()` return by reference.
- Alter `__get()` to raise an `E_USER_DEPRECATED`, and then proxy to the appropriate getter.

If you do those changes, I will merge.